classes/controller/ProductListingFrontController.php

```diff
@@ -392,19 +392,37 @@ protected function getProductSearchVariables()
     }
 
     /**
-     * @param FacetCollection $facetCollection
+     * @param ProductSearchResult $result
      */
-    protected function filterFacetCollection(FacetCollection $facetCollection)
+    protected function applyCatalogMode(ProductSearchResult $result)
     {
-        $filteredFacets = [];
-        /** @var Facet $facet */
-        foreach ($facetCollection->getFacets() as $facet) {
-            if (Configuration::get('PS_CATALOG_MODE') && 'price' === $facet->getType()) {
-                continue;
+        if (!Configuration::get('PS_CATALOG_MODE')) {
+            return;
+        }
+
+        if ($result->getFacetCollection()) {
+            $filteredFacets = [];
+            /** @var Facet $facet */
+            foreach ($result->getFacetCollection()->getFacets() as $facet) {
+                if (Configuration::get('PS_CATALOG_MODE') && 'price' === $facet->getType()) {
```

Comment:
Why `Configuration::get('PS_CATALOG_MODE') ` when you check for `!Configuration::get('PS_CATALOG_MODE') ` on line 399 :)

Reply:
oups you're right, some old code I forgot to remove!

Reply:
done!